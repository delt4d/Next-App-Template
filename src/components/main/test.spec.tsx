import { render, screen } from '@testing-library/react'
import { ReactNode } from 'react'

import Main from '.'

const TestWrapper = ({ children }: { children: ReactNode }) => {
    return <div className='h-screen'>{children}</div>
}

const renderMain = () => render(<Main />, { wrapper: TestWrapper })

describe('<Main />', () => {
    it('should render the heading', () => {
        renderMain()
        expect(screen.getByRole('heading', { name: /next app/i }))
    })

    it('should match snapshot', () => {
        const { container } = renderMain()
        expect(container.firstChild).toMatchSnapshot()
    })
})
