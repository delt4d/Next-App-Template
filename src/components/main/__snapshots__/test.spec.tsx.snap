// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshot 1`] = `
<div
  class="h-screen"
>
  <main
    class="undefined flex items-center justify-center"
  >
    <h1
      class="title mb-2"
    >
      Next App
    </h1>
  </main>
</div>
`;
